src/mgr/ActivePyModules.cc

```diff
@@ -99,13 +99,13 @@ PyObject *ActivePyModules::get_server_python(const std::string &hostname)
 PyObject *ActivePyModules::list_servers_python()
 {
   PyThreadState *tstate = PyEval_SaveThread();
-  Mutex::Locker l(lock);
-  PyEval_RestoreThread(tstate);
   dout(10) << " >" << dendl;
 
   PyFormatter f(false, true);
```

Comment:
This is wrong, as we can't touch PyFormatter without holding the GIL, which as a result can cause various GC problems.

This has been fixed in https://github.com/ceph/ceph/pull/26445 by Sage.